client/reader/sidebar/index.jsx

```diff
@@ -254,22 +247,37 @@ const ReaderSidebar = React.createClass( {
 					onTagExists={ this.highlightNewTag }
 					currentTag={ this.state.currentTag } />
 
-				</SidebarRegion>
+			</SidebarRegion>
 
-				{ this.renderAppPromo() }
+			{ this.shouldRenderAppPromo() &&
+				<div className="reader-sidebar__AppPromo">
+					<AppPromo location="reader" locale={ userUtils.getLocaleSlug() } />
+				</div>
+			}
 
 				<SidebarFooter />
 			</Sidebar>
 		);
 	}
 } );
 
+ReaderSidebar.defaultProps = {
+	translate: identity
+};
+
+export { ReaderSidebar };
 export default connect(
 	( state ) => {
 		return {
 			isListsOpen: state.ui.reader.sidebar.isListsOpen,
 			isTagsOpen: state.ui.reader.sidebar.isTagsOpen,
-			subscribedLists: getSubscribedLists( state )
+			subscribedLists: getSubscribedLists( state ),
+			isDesktopPromoDisabled: store.get( 'desktop_promo_disabled' ),
+			isViewportMobile: viewport.isMobile(),
+			isUserLocaleEnglish: 'en' === userUtils.getLocaleSlug(),
+			isChromeOS: isChromeOS(),
+			isDesktopPromoConfiguredToRun: config.isEnabled( 'desktop-promo' ),
+			isUserDesktopAppUser: userSettings.getSetting( 'is_desktop_app_user' )
```

Comment:
not sure I understand. does the reader sidebar really care if we are in ChromeOS or if it's a desktop user? couldn't we flip those bits in a function outside of the component and keep the prop list pretty slim?